bin/yb-ctl

```diff
@@ -1402,11 +1537,18 @@ class ClusterControl:
         self.creating_cluster = True
         server_counts = self.options.replication_factor
         self.set_master_addresses()
-        self.cluster_config = {
-            "enable_ysql": self.options.enable_ysql,
-            "num_drives": self.options.num_drives,
-            "replication_factor": self.options.replication_factor
-        }
+        self.cluster_config = {}
+        self.options.set_cluster_config(self.cluster_config)
+
+        for attr_name in (
+            "enable_ysql",
+            "num_drives",
+            "replication_factor"
+        ):
+            attr_value = getattr(self.options, attr_name)
+            if attr_value is not None:
+                self.cluster_config[attr_name] = attr_value
+        self.cluster_config['ports'] = self.options.base_ports
```

Comment:
hmm, L772: `        if 'base_ports' in cluster_config:`, shouldn't this be `self.cluster_config['base_ports']` ?

Reply:
This is a valid dictionary key existence test in Python.

Reply:
oh, sry, I can see how that might have been confusing.

I mean here, in L1551, should we not be setting self.cluster_config['base_ports'], since further up in L772, we're checking for `base_ports`, not for `ports`...

Reply:
OK, good point, fixed.